src/algebra/order/monoid_lemmas_gt_zero.lean

```diff
@@ -0,0 +1,188 @@
+  being in a subtype of `α`.
+
+We combine this with (`contravariant_`) `covariant_class`es to assume that multiplication by
+positive elements is (strictly) monotone on a `mul_zero_class`, `monoid_with_zero`,...
+More specifically, we use extensively the following typeclasses:
+
+* monotone left
+* * `covariant_class α>0 α pos_mul (≤)`, abbreviated `pos_mul_mono α`,
+    expressing that multiplication by positive elements on the left is monotone;
+* * `covariant_class α>0 α pos_mul (<)`, abbreviated `pos_mul_strict_mono α`,
+    expressing that multiplication by positive elements on the left is strictly monotone;
+* monotone right
+* * `covariant_class α>0 α mul_pos (≤)`, abbreviated `mul_pos_mono α`,
+    expressing that multiplication by positive elements on the right is monotone;
+* * `covariant_class α>0 α mul_pos (<)`, abbreviated `mul_pos_strict_mono α`,
+    expressing that multiplication by positive elements on the right is strictly monotone.
+* reverse monotone left
+* * `contravariant_class α>0 α pos_mul (≤)`, abbreviated `pos_mul_mono_rev α`,
+    expressing that multiplication by positive elements on the left is reverse monotone;
+* * `contravariant_class α>0 α pos_mul (<)`, abbreviated `pos_mul_strict_mono_rev α`,
+    expressing that multiplication by positive elements on the left is strictly reverse monotone;
+* reverse reverse monotone right
+* * `contravariant_class α>0 α mul_pos (≤)`, abbreviated `mul_pos_mono_rev α`,
+    expressing that multiplication by positive elements on the right is reverse monotone;
+* * `contravariant_class α>0 α mul_pos (<)`, abbreviated `mul_pos_strict_mono_rev α`,
+    expressing that multiplication by positive elements on the right is strictly reverse monotone.
+
+##  Formalization comments
+
+We use `α>0 = {x : α // 0 < x}` with a strict inequality since in most cases what happens with `0`
+is clear.  This creates a few bumps in the first couple of proofs, where we have to split cases on
+whether an element is `0` or not, but goes smoothly after that.  A further advantage is that we
+only introduce notation for the positive elements and we do not need also the non-negative ones.
+-/
+
+/- I am changing the file `algebra/order/monoid_lemmas` incrementally, with the idea of
+reproducing almost all of the proofs in `algebra/order/ring` with weaker assumptions. -/
+
+universe u
+variable {α : Type u}
+
+/-  Notation for positive elements
+https://
+leanprover.zulipchat.com/#narrow/stream/113488-general/topic/notation.20for.20positive.20elements
+-/
+local notation `α>0` := {x : α // 0 < x}
+
+namespace zero_lt
+
+/--
+`pos_mul` is the multiplication of an element of the subtype `α>0 = {x : α // 0 < x}` of positive
+elements by an element of the type itself.  The element of the subtype appears on the left:
+`pos_mul a b = a * b`.
+
+`mul_pos` is the multiplication in the other order. -/
+@[protected]
+private def pos_mul [has_zero α] [has_lt α] [has_mul α] : α>0 → α → α :=
```

Comment:
Whatever we do, absolutely do **not** make these definitions private. If they ever appear in a goal for some reason, it means you cannot even `dsimp [pos_mul]` them in a different file.

Then again, I still don't like having these definitions. These names are no good, they sound too much like `mul_pos` (in the root namespace), and it sounds like they're a lemma.